src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -352,11 +352,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundArguments.Add(boundArgument)
 
                 Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasErrors)
-                elementTypes.Add(elementType)
-
                 If elementType Is Nothing Then
                     hasNaturalType = False
+                    elementType = ReclassifyExpression(boundArgument, diagnostics).Type
```

Comment:
>ReclassifyExpression(boundArgument, diagnostics) [](start = 34, length = 48)

It feels like this could result in all sorts of premature diagnostics.